src/components/ActivityForm/ActivityForm.js

```diff
@@ -101,6 +105,10 @@ export default class ActivityForm extends React.Component {
       if (!_.isEmpty(errors)) {
         this.setState(errors);
       }
+    } else if (validateFreeTimeSlot({ day, activityOrdinal }, currentActivities) === false) {
```

Comment:
1. do we need this `else` ? maybe couple of errors can occure at the same time. 
2. (!validateFreeTime...) {}